### PR TITLE
[v12] Metrics: add IsSSO to Discover Events

### PR DESCRIPTION
**Pull request by marcoandredinis**
Backport #23791 to branch/v12

### CODE REVIEW

**Review by Tener — APPROVED** [review]
Do we truly need to synchronize the changes? I would expect that forward-compatible changes like these (i.e., adding a single field in proto) would be possible to be deployed without synchronization, as this was one of the use cases behind the design of protobuf.

**Comment by marcoandredinis** [comment]
> Do we truly need to synchronize the changes? I would expect that forward-compatible changes like these (i.e., adding a single field in proto) would be possible to be deployed without synchronization, as this was one of the use cases behind the design of protobuf.

Yeah, I think so.
The worst that can happen is the field not being stored in Posthog.

@espadolini Do you see any problem with merging now?
This adds a new field to the DiscoverMetadata (used in multiple events)

**Comment by espadolini** [comment]
Should be fine, the field is just going to not exist from the server's point of view.

IMO it's still important from a process standpoint to wait until the proto is merged in cloud _master_ before merging in teleport master so we don't accidentally add different fields or messages with the same name/number (which would be bad), but if you're ok with missing data because the server doesn't know about some fields, it's fine to merge in teleport releases before the server side is deployed in cloud prod.

**Review by smallinsky — APPROVED** [review]

**Pull request merged**
